File: Python/3-primes.py
import math

def primes(n):
	number = 0
	count = 0
	while count < n:
		while True:
			if is_prime(number):
				yield number
				count += 1
				number += 1
				break
			number += 1

def is_prime(number):
    if number > 1:
        if number == 2:
            return True
        if number % 2 == 0:
            return False
        for current in range(3, int(math.sqrt(number) + 1), 2):
            if number % current == 0: 
                return False
        return True
    return False

for i in primes(10):
	print(i)

File: JavaScript/3-primes.js
'use strict';

function* numbers(start) {
  while (true) {
    yield start++;
  }
}

function* sieve(sequence, prime) {
  for (const number of sequence) {
    if (number % prime !== 0) {
      yield number;
    }
  }
}

function* primes() {
  let sequence = numbers(2);
  let prime;
  while (true) {
    prime = sequence.next().value;
    yield prime;
    sequence = sieve(sequence, prime);
  }
}

function* take(count, sequence) {
  for (let i = 0; i < count; i++) {
    yield sequence.next().value;
  }
}

for (const prime of take(10, primes())) {
  console.log(prime);
}
